BugFix: fix corner case where the lua interpreter is `lua5.1` on Windows

SlySven opened the pull request:
The '.1' is taken as the extension on that OS in some situations instead of '.exe' if the latter is not explicitly included in the invocation.

This is part of my work toward migrating to MSYS2 builds on the Windows platform.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: fix corner case where the lua interpreter is `lua5.1` on Windows

The '.1' is taken as the extension on that OS in some situations instead
of '.exe' if the latter is not explicitly included in the invocation.

This is part of my work toward migrating to MSYS2 builds on the Windows
platform.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on translations/translated/updateqm.pri:
```diff
@@ -41,7 +41,8 @@ for(file, TS_FILES_NOEXT) {
 }
-STATS_GENERATOR = $$shell_path("$${PWD}/generate-translation-stats.lua")
+STATS_GENERATOR = $$system_path("$${PWD}/generate-translation-stats.lua")
+message("Will use \"$${STATS_GENERATOR}\" to parse translation statistics...")
```
Could you take this out now that it's working? It's better to have less messages, not more messages that you adapt to ignore.

SlySven replied:
Would you accept them being just commented out? So an interested party (which, given the weird places {cygwin/FreeBSD/MSYS2} I occasionally come up with to build Mudlet will probably be me anyhow) can quickly re-enable to check - rather than having to remember what I used?

vadi2 replied:
Yeah, I think that sounds good!

_maybe we could do the same with a few others, too_...

SlySven replied:
We shouldn't forget that it is run for all builds not just the CI ones.

vadi2 replied:
Yeah, it's the all builds that it's a problem, it has a lot of unnecessary information 99% of the time that you ignore. Like unix tools, one only needs to see the errors, not the successful operation.

vadi2 commented on translations/translated/updateqm.pri:
```diff
@@ -64,7 +65,15 @@ isEmpty(LUA_SEARCH_OUT) {
+    }
 }
 
+message("Running: $$LUA_COMMAND $$STATS_GENERATOR ...")
```
Same here.

vadi2 commented on translations/translated/updateqm.pri:
```diff
@@ -64,7 +65,15 @@ isEmpty(LUA_SEARCH_OUT) {
+        # The '.' in the file name confuses the QMake system() utility or the
+        # shell used in the Windows case as it is considered to be an extension
+        # unless we put in the actual 'exe' extension as well:
+        LUA_COMMAND = "lua5.1.exe"
```
I'm not sure what are we supposed to do with the comment this information provides?

SlySven replied:
I should have said `.1` there admittedly - and it was the explanation why we do have `LUA_COMMAND = "lua5.1.exe"` rather than `LUA_COMMAND = "lua5.1"` - as I say, it caught me out with MSYS2 as that IS the executable that is used to start a lua interpreter on the command line.

vadi2 replied:
Aha! That makes sense.

SlySven replied:
Curiously `lua5.1` is also the one that gets used by the Travis CI used on macOs as well...

vadi2 reviewed (commented):
Simple change, looks good :+1:

SlySven committed:
Revise: adopt some peer review suggestions

This includes commenting out some debugging output lines now that things
*seem* to be working.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
